### PR TITLE
Users addtion and login

### LINKED ISSUE #5: Login web
# REQUIREMENTS
Allows the administrator and users with access to reports to enter the web application.

# DESIGN
Does not apply

# Acceptance Criteria
- [x] Implement a sistem of login with devise gems
- [x] Access key reset mechanisms
- [x] Integration of the modules created before of close  the system

### LINKED ISSUE #6: Users Crud
# REQUIREMENTS
This module will only be available to the administrator.
When registering a new user, the application will send an email notifying your registration and     the access code that you can use to enter.

The data required for this module are: 
Número de control 
Nombre 
Correo electrónico (servirá como identificador de usuario al momento de iniciar sesión) 
Clave de acceso 
Institución (seleccionar del catálogo o agregar uno nuevo) 
Departamento (seleccionar del catálogo o agregar uno nuevo) 
Acceso a reportes (El administrador de permisos para que los usuarios accedan a los reportes)

# DESIGN
Does not apply

# Acceptance Criteria
- [x] Created users
- [x] Show users
- [x] Updated users
- [x] Deleted users

### LINKED ISSUE #19: Authentication by API
# REQUIREMENTS
Write a detailed description for this issue, provide all necessary information so any member of the project understands
- Authentication user using JWT
- Protect endpoints if user is not authenticated or if  token expired.

# DESIGN
No Apply

# Acceptance Criteria
List all required acceptance criteria
- [x] Protected endpoints
- [x] Auth User
- [x] Get data user
- [x] Unit testing

### PR DESCRIPTION
**Pull request by ramasoftdev**
# Description
What did you implemented/Why did you implemented this:
 - This PR adds devise for login and implements roles and permission by its.
 - Also is added Bootstrap 4 for a better view handling.
 - The tests files were maded, but missing the correct implementation.
 - The ticket for referent about this PR is [LINK](https://github.com/BC-NOV-20-Proyectos/alerta-covid-web-team01/issues/5).
 
 # Testing
 Missing yet tests, But you can check out the following files:
1. app/controllers/application_controller.rb
2. app/controllers/roles_controller.rb
3. app/controllers/users_controller.rb

4. app/helpers/roles_helper.rb

5. app/models/ability.rb
6. app/models/permission.rb
7. app/models/role.rb
8. app/models/user.rb

9. app/views/layouts/_navbar.html.erb
10. app/views/roles/edit.html.erb
11. app/views/roles/index.html.erb
12. app/views/roles/show.html.erb
13. app/views/users/index.html.erb

14. config/routes.rb

15. db/migrate/20210413230916_create_roles.rb
16. db/migrate/20210413231305_add_role_id_to_users.rb
17. db/migrate/20210413231714_create_permissions.rb
18. db/migrate/20210414015335_create_permissions_roles.rb

### CODE REVIEW

**Review by ArielMD — APPROVED** [review]
wow, that looks good, but there are a lot of files xd

File: alerta-covid/app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+// app/javascript/stylesheets/application.scss
```
Comment (mariormt17) [inline alerta-covid/app/assets/stylesheets/application.scss:1]: You can delete these empty stylesheets files

File: alerta-covid/app/helpers/roles_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module RolesHelper
```
Comment (mariormt17) [inline alerta-covid/app/helpers/roles_helper.rb:1]: And you can delete these empty helper files too

**Review by mariormt17 — APPROVED** [review]
Very good job! It is a breakthrough for the project

**Review by FranckZnt19 — APPROVED** [review]
Good work, excellent!

**Pull request merged**